crates/building_blocks_storage/src/vox_format_conversions.rs

```diff
@@ -0,0 +1,148 @@
+use std::{fs::File, io::{Cursor, Read, Seek}, path::Path};
+
+use crate::{
+    Array3x1,
+    GetMut,
+};
+use vox_format::{
+    data::{VoxBuffer, VoxModelBuffer}, 
+    reader::{Error, read_vox_into}, 
+    types::{Color, ColorIndex, Palette, Point, Size, Voxel},
+};
+
+use building_blocks_core::{Extent3i, Point3i, PointN};
+    
+/// Re-export `vox_format` crate.
+pub use vox_format;
+
+
+/*
+
+These must live in building_blocks_core`
+
+impl<T> From<Vector<T>> for PointN<[T; 3]> {
+    fn from(v: Vector<T>) -> Self {
+        PointN(v.into())
+    }
+}
+
+impl From<Size> for Extent3i {
+    fn from(size: Size) -> Self {
+        // Note: This can fail, if the component is greater than `i32::MAX`
+        Extent3i::from_min_and_shape(
+            Default::default(),
+            PointN([size.x as i32, size.y as i32, size.z as i32]),
+        )
+    }
+}
+
+impl From<Point> for Point3i {
+    fn from(point: Point) -> Self {
+        PointN([point.x as i32, point.y as i32, point.z as i32])
+    }
+}
+*/
+
+
+fn point3i_from_vox(point: Point) -> Point3i {
+    PointN([point.x as i32, point.y as i32, point.z as i32])
+}
+
+fn extent3i_from_vox(size: Size) -> Extent3i {
+    // Note: This can fail, if the component is greater than `i32::MAX`
+    Extent3i::from_min_and_shape(
+        Default::default(),
+        PointN([size.x as i32, size.y as i32, size.z as i32]),
+    )
+}
+
+
+impl VoxModelBuffer for Array3x1<ColorIndex> {
+    fn new(size: Size) -> Self {
+        Array3x1::fill_with(extent3i_from_vox(size), |_point| ColorIndex::default())
```

Comment:
This could be faster if you fill the channel first, like:

```rust
let extent = extent3i_from_vox(size);
let channel = Channel::fill(extent.num_points(), ColorIndex::default());
Array3x1::new(extent, channel)
```

I should probably just have an `Array::fill` function.

Reply:
With the new `VoxChannel` trait I don't want to impose a requirement for `VoxChannel: Clone`. Shouldn't this compile down to the same code anyway?